app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,20 @@
+class MoviesController < ApplicationController
+
+  def show
+    @movie = Movie.find(params[:id])
+    @actor_names = @movie.order_by_age
+    @avg_age = @movie.avg_age
+    if params[:search_by_name].present? 
+      threshold = params[:search_by_name].to_s
+      @actors = Actor.filter_by_name(threshold)
+    end
+  end
+
+  def update
+    @movie = Movie.find(params[:id])
+    @actor = Actor.find(params[:actor_id])
```

Comment:
Your form implementation didn't pass an `actor_id` because your form was looking for a name. So this line probably couldn't execute properly. Your form should have passed the actor ID via the form field